setup/nextcloud.sh

```diff
@@ -206,39 +215,38 @@ if [ ! -f $STORAGE_ROOT/owncloud/owncloud.db ]; then
 
 	# Create an initial configuration file.
 	instanceid=oc$(echo $PRIMARY_HOSTNAME | sha1sum | fold -w 10 | head -n 1)
-	cat > $STORAGE_ROOT/owncloud/config.php <<EOF;
-<?php
-\$CONFIG = array (
-  'datadirectory' => '$STORAGE_ROOT/owncloud',
-
-  'instanceid' => '$instanceid',
-
-  'forcessl' => true, # if unset/false, Nextcloud sends a HSTS=0 header, which conflicts with nginx config
-
-  'overwritewebroot' => '/cloud',
-  'overwrite.cli.url' => '/cloud',
-  'user_backends' => array(
-    array(
-      'class' => '\OCA\UserExternal\IMAP',
-      'arguments' => array(
-        '127.0.0.1', 143, null, null, false, false
-       ),
-    ),
-  ),
-  'memcache.local' => '\OC\Memcache\APCu',
-  'mail_smtpmode' => 'sendmail',
-  'mail_smtpsecure' => '',
-  'mail_smtpauthtype' => 'LOGIN',
-  'mail_smtpauth' => false,
-  'mail_smtphost' => '',
-  'mail_smtpport' => '',
-  'mail_smtpname' => '',
-  'mail_smtppassword' => '',
-  'mail_from_address' => 'owncloud',
-);
-?>
+	CONFIG_TEMP=/tmp/cfg-$instanceid.json
+	cat > $CONFIG_TEMP <<EOF
+	{
+		"system": {
+			"datadirectory": "$STORAGE_ROOT/owncloud",
+			"instanceid": "$instanceid",
+			"forcessl": true,
+			"overwritewebroot": "/cloud",
+			"overwrite.cli.url": "https://${PRIMARY_HOSTNAME}/cloud",
+			"user_backends": [
+				{
+					"class": "\\\\OCA\\\\UserExternal\\\\IMAP",
+					"arguments": [ "127.0.0.1", 143, null, null, false, false ]
+				}
+			],
+		 	"memcache.local": "\\\\OC\\\\Memcache\\\\APCu",
+		 	"mail_smtpmode": "sendmail",
+		 	"mail_smtpsecure": "",
+		 	"mail_smtpauthtype": "LOGIN",
+		 	"mail_smtpauth": false,
+		 	"mail_smtphost": "",
+		 	"mail_smtpport": "",
+			"mail_smtpname": "",
+			"mail_smtppassword": "",
+			"mail_from_address": "owncloud"
+		}
+	}
 EOF
 
+	sudo -u www-data php8.0 /usr/local/lib/owncloud/occ config:import $CONFIG_TEMP
```

Comment:
Use php$PHP_VER instead of php8.0? Also on line 355

Reply:
thank you. fixed.